src/configCreator.ts

```diff
@@ -0,0 +1,68 @@
+import { CLIEngine, Linter } from "eslint"
+import { defaultOptions } from "./eslintDefaultOptions"
+import { Codacyrc, Pattern, ParameterValue } from "./model/CodacyInput"
+import { toolName } from "./toolMetadata"
+import { patternIdToEslint } from "./model/Patterns"
+import { cloneDeep, isEmpty, partition, fromPairs } from "lodash"
+import { allFilesNames } from "./allFiles"
+
+function patternsToRules(
+  patterns: Pattern[]
+): { [name: string]: Linter.RuleLevel | Linter.RuleLevelAndOptions } {
+  const pairs = patterns.map(pattern => {
+    const patternId = patternIdToEslint(pattern.patternId)
+    if (pattern.parameters) {
+      const [unnamedParameters, namedParameters] = partition(
+        pattern.parameters,
+        p => p.name === "unnamedParam"
+      )
+      const namedOptions = fromPairs(
+        namedParameters.map(p => [p.name, p.value])
+      )
+      const unnamedOptions = unnamedParameters.map(p => p.value)
+      return [
+        patternId,
+        isEmpty(namedOptions)
+          ? ["error", ...unnamedOptions]
+          : ["error", ...unnamedOptions, namedOptions]
+      ]
+    } else {
+      return [patternId, "error"]
+    }
+  })
+
+  return fromPairs(pairs)
+}
+
+async function createOptions(
+  srcDirPath: string,
+  codacyInput?: Codacyrc
+): Promise<CLIEngine.Options> {
+  if (codacyInput && codacyInput.tools) {
+    const eslintTool = codacyInput.tools.find(tool => tool.name === toolName)
+    if (eslintTool && eslintTool.patterns) {
+      const patterns = eslintTool.patterns
+      const result = cloneDeep(defaultOptions)
+      if (result.baseConfig) {
+        result.baseConfig.extends = [] // TODO: Maintain base configurations without rules
+        result.baseConfig.overrides.extends = []
+        result.baseConfig.rules = patternsToRules(patterns)
+      }
+      result.useEslintrc = false
+      return result
+    }
+  }
+  const fileNames = await allFilesNames(srcDirPath)
+  if (fileNames.find(name => name.startsWith(".eslintrc")) !== undefined)
```

Comment:
here I think you can get a wrong `.eslintrc` 
e.g. from a subfolder that can be a dead file in the repo

Reply:
Here the problem is tricky.
Because Eslint support `.eslintrc` files in any subfolder.
Do you have suggestions here?

Reply:
You cannot search for the file. ESLint needs to find the files themselves when the user wants config file.
It needs to support multiples files in the tree, you don't want to do that. Please read the old code before doing this to make sure you cover all the cases so we don't introduce bugs again.